### ISSUE TITLE
Error on 25% of Win7/8 machines: [System.Net.SecurityProtocolType] does not con tain a method named 'op_Addition'

### ISSUE THREAD

**Issue by StephenWeatherford** [dotnet/vscode-dotnet-runtime#64]

**Comment by bozturkMSFT** [comment]
This should be reproducible on any machine with a Powershell that is using .Net Framework version lower than 4.6.

Issue should be on [this line](https://github.com/sfoslund/vscode-dotnetcore/blob/1eeed3b7366039b71d93588e080417baff39ee91/vscode-dotnet-runtime-library/src/Acquisition/AcquisitionInvoker.ts#L37).
I don't remember the reason behind it, but the error shouldn't happen if additions are done separately like this:
```
[System.Net.ServicePointManager]::SecurityProtocol = [System.Net.SecurityProtocolType]::Tls12;
[System.Net.ServicePointManager]::SecurityProtocol += [System.Net.SecurityProtocolType]::Tls13;
```

**Comment by sfoslund** [comment]
@TylerLeonhardt suggested that the following should fix the issue as well:
```
[Net.ServicePointManager]::SecurityProtocol = [Net.ServicePointManager]::SecurityProtocol -bor [Net.SecurityProtocolType]::Tls12
```

**Comment by TylerLeonhardt** [comment]
Since .NET Core already fixed this is all this works in PowerShell 6+, make sure you only run that for Windows PowerShell.

You can use `$IsCoreCLR` which is only true in PS6+ (and defined in Windows PowerShell 5.1), or look at `$PSVersionTable`.

**Comment by sfoslund** [comment]
@TylerLeonhardt okay, so something like this?:
```
powershell.exe -ExecutionPolicy unrestricted -Command "& { If (-not $IsCoreCLR) { [Net.ServicePointManager]::SecurityProtocol = [Net.ServicePointManager]::SecurityProtocol -bor [Net.SecurityProtocolType]::Tls12 } ; & ${installCommand} }
```

**Comment by TylerLeonhardt** [comment]
Well I guess if you're always executing in powershell.exe, that will always be Windows PowerShell (PS6+ is pwsh.exe) so I guess you don't really need to if statement.

Is there a reason that line is in the C# directly instead of being in the install script?

**Comment by sfoslund** [comment]
The install script is in powershell and here we're running on whatever the default shell happens to be. This command isn't broadly applicable to users who are directly working with the install script, but since we're dynamically downloading the script and running it in the background we need it here specifically. 

Okay, great, if this looks good to you I'll make a PR.
> powershell.exe -ExecutionPolicy unrestricted -Command "& { [Net.ServicePointManager]::SecurityProtocol = [Net.ServicePointManager]::SecurityProtocol -bor [Net.SecurityProtocolType]::Tls12 ; & ${installCommand} }

**Comment by TylerLeonhardt** [comment]
Wouldn't any user who is using Windows PowerShell with .NET Framework 4.6 or lower need this command though?

**Comment by sfoslund** [comment]
Yes, and there was an attempt to add a similar line directly to the script, but it backfired in ways I don't understand (see [this change](https://github.com/dotnet/website/pull/1877) that later got backed out). @bozturkMSFT (whose team owns the script) would have a better explanation. 

Just made a PR for this fix: #66. Thanks a lot for your expertise, @TylerLeonhardt!

**Comment by TylerLeonhardt** [comment]
Thanks for the context, @sfoslund! Looks like the same change so my guess is that it would throw this same exceprtion...

I'm just gonna copy what I mentioned in email here so @bozturkMSFT can see!:

"I just want to be clear that this isn't really a PowerShell issue since all the code involved here is just making calls to .NET.

The error:
```
[System.Net.SecurityProtocolType] does not contain a method named 'op_Addition'
 ```
Comes from .NET. I would say that maybe the docs had something on this type not supporting addition:

https://docs.microsoft.com/en-us/dotnet/api/system.net.securityprotocoltype?view=netframework-4.5

But unfortunately they don't...

Maybe `FlagAttribute` types didn't have addition support in .NET Framework 4.6 and lower? I'm not sure.

Anyway, I hope that helps. PowerShell is just another .NET language in this issue, I think.
"
